src/NServiceBus.AmazonSQS/MessagePump.cs

```diff
@@ -137,8 +137,13 @@ public async Task Stop()
             cancellationTokenSource?.Cancel();
 
             await Task.WhenAll(pumpTasks).ConfigureAwait(false);
-
             pumpTasks?.Clear();
+            
+            while (maxConcurrencySempahore.CurrentCount != maxConcurrency)
```

Comment:
if at this point all the pump tasks are gone what are we waiting for?

Reply:
Messages in flight that are about to complete. So essentially they passed the semaphore and might already called the core

Reply:
got it thanks.